### PR TITLE
Add missing examples and questions

### PR DESCRIPTION
**Pull request by przemyslawzalewski**
This PR brings examples and questions for every module except the date module - I will tackle with tests, examples, and questions for that in a separate PR at once.
Example code is now automatically formatted with prettier with reduced print width to nicely fit on mobile 🎉

I have already deployed the docs so you can preview the new examples and questions at https://sandstreamdev.github.io/std/

Please review commit-by-commit - the last commit includes only the regenerated code.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2094,6 +3716,19 @@ Searches the given object by the given predicate and returns the found entry or
...
+<!-- prettier-ignore-start -->
+```javascript
+findEntry(({ x }) => x % 2 === 0)({ a: { x: 1 }, b: { x: 2 }, c: { x: 3 } });
+// ⇒ ["b", { x: 2 }]
+```
+<!-- prettier-ignore-end -->
+
+##### Questions
+
+- How to find an entry of an object by a function?
```
Comment (marekrozmus) [inline README.md]: `by` => `with` ?

File: README.md [hunk 2]
```diff
@@ -2551,6 +4559,31 @@ Checks if the given string is present and is not empty or all whitespace.
...
+nonEmpty(null);
+// ⇒ false
+```
+<!-- prettier-ignore-end -->
+
+##### Questions
+
+- How to check if a string is non empty?
+- How to check is string is not all whitespace?
+- How to check is string is not null or undefined?
```
Comment (marekrozmus) [inline README.md]: `is` => `if` in both lines

File: README.md [hunk 3]
```diff
@@ -2614,6 +4691,25 @@ Calculates a cross product of the given vectors. Returns a scalar.
...
+```javascript
+cross([3, 5], [-1, -8]);
+// ⇒ -19
+```
+<!-- prettier-ignore-end -->
+
+##### Questions
+
+- How to compute a cross product of two vectors?
+- How to check on which side of a line point is?
```
Comment (marekrozmus) [inline README.md]: `the` point is ??

File: README.md [hunk 4]
```diff
@@ -2857,6 +5109,19 @@ Composes class name from truthy values with support of string and objects.
...
+<!-- prettier-ignore-start -->
+```javascript
+classNames("test", { active: true, disabled: false, on: undefined });
+// ⇒ "test active"
+```
+<!-- prettier-ignore-end -->
+
+##### Questions
+
+- How to create a class name from list of string and/or objects?
```
Comment (marekrozmus) [inline README.md]: `string` => `strings` ?

File: README.md [hunk 5]
```diff
@@ -2874,6 +5139,23 @@ Stops propagation and prevents the default handler of the given event.
...
+  stopPropagation: () => console.log("stopPropagation")
+};
+
+cancel(event);
+```
+<!-- prettier-ignore-end -->
+
+###### Questions
+
+- How to stop propagation and prevent default at once?
```
Comment (marekrozmus) [inline README.md]: ... to stop `event` propagation

File: README.md [hunk 6]
```diff
@@ -303,6 +565,24 @@ Checks if lengths of given arrays differ.
...
+
+```javascript
+lengthDiffers([6, 7], [1, 2]);
+// ⇒ false
+```
+<!-- prettier-ignore-end -->
+
+##### Questions
+
+- How to check if array lengths differ?
```
[the remark this answers is not in the document]
Comment (przemyslawzalewski) [reply README.md]: I will add a separate question :)

File: README.md (hunk 1 above)
Comment (przemyslawzalewski) [reply README.md]: I have clarified that it is about a `predicate function` and kept the `by` form.

File: README.md [hunk 7]
```diff
@@ -327,66 +625,182 @@ Returns the middle element or the right one when the number of elements is even.
...
+midpoint([1, 2, 3, 4]);
+// ⇒ 3
 ```
 <!-- prettier-ignore-end -->
 
-#### multiple
+##### Questions
 
-Checks if the given array contains more than one element.
+- How to get the element in the middle of an array?
```
[the remark this answers is not in the document]
Comment (przemyslawzalewski) [reply README.md]: Seems to be a proper form, e.g. https://www.quora.com/How-can-I-add-new-elements-in-the-middle-of-an-existing-array

[the remark this answers is not in the document]
Comment (przemyslawzalewski) [reply README.md]: I have also added "How to get the middle element of an array?".

File: README.md (hunk 3 above)
Comment (przemyslawzalewski) [reply README.md]: I have added "a" instead of "the".

File: README.md (hunk 2 above)
Comment (przemyslawzalewski) [reply README.md]: Thanks!

File: README.md [hunk 8]
```diff
@@ -1753,18 +2991,44 @@ Calculates the average of given array of numbers.
...
+<!-- prettier-ignore-start -->
+```javascript
+ceilToNearestPowerOfTwo(345);
+// ⇒ 512
 ```
 <!-- prettier-ignore-end -->
 
+##### Questions
+
+- How to get the nearest power of two greater or equal to the given value/
```
[the remark this answers is not in the document]
Comment (przemyslawzalewski) [reply README.md]: You're right. A typo. Thanks!

File: README.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (marekrozmus) [reply README.md]: > Seems to be a proper form, e.g. https://www.quora.com/How-can-I-add-new-elements-in-the-middle-of-an-existing-array

But isn't it about adding new one, not getting existing one?

File: array/unique.json [hunk 9]
```diff
@@ -1,12 +1,12 @@
...
   "signature": "(xs: any[]) => any[]",
   "examples": [
     {
       "language": "javascript",
-      "content": "unique(); // ⇒ TODO"
+      "content": "unique([1, 2, 3, 4, 3, 4, 3, 6]);\n// ⇒ [1, 2, 3, 4, 6]"
     }
   ],
-  "questions": ["TODO: List questions that may this function answers."]
+  "questions": ["How to get all unique values in an array?"]
```
Comment (marekrozmus) [inline array/unique.json]: Again, I'm not sure if `get ... in` is proper verb for saying that we take something out of array. https://www.oxfordlearnersdictionaries.com/definition/english/get-in

File: async/README.md [hunk 10]
```diff
@@ -1,6 +1,6 @@
 # debounce
 
-Makes the function run after the given period of not being called. Useful to delay input submission for autocomplete etc.
+Makes the function run after the given period of not being called. Useful to delay input submission for Makes the function run after the given period of not being called. Useful to delay input submission for auto-complete etc.
```
Comment (marekrozmus) [inline async/README.md]: what just happened here :)?

File: debug/assert.test.ts [hunk 11]
```diff
@@ -26,7 +26,7 @@ describe("assert", () => {
     expect(() => assert(false, "Custom message")).toThrow("Custom message");
   });
 
-  it("calls a callback if given", () => {
+  it("calls a callback if the given", () => {
```
Comment (marekrozmus) [inline debug/assert.test.ts]: `if the given` what? It was ok.

File: object/fromEntries.json [hunk 12]
```diff
@@ -1,12 +1,15 @@
 {
   "name": "fromEntries",
-  "description": "Creates an object from array of key value pairs (entries).",
+  "description": "Creates an object from an array  of key-value pairs (entries).",
```
Comment (marekrozmus) [inline object/fromEntries.json]: extra space before `of`

File: object/fromEntries.json [hunk 13]
```diff
@@ -1,12 +1,15 @@
...
     {
       "language": "javascript",
-      "content": "fromEntries(); // ⇒ TODO"
+      "content": "fromEntries([[\"a\", 1], [\"b\", 2], [\"c\", 3]]);\n// ⇒ { a: 1, b: 2, c: 3 }"
     }
   ],
-  "questions": ["TODO: List questions that may this function answers."]
+  "questions": [
+    "How to create an object from an array of key-value pairs?",
+    "How to create an object from an array  of entries?"
```
Comment (marekrozmus) [inline object/fromEntries.json]: extra space before `of`

File: README.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (marekrozmus) [reply README.md]: As I'm looking at my comment then I see that I messed up something but the second question looks great :)

File: array/unique.json (hunk 9 above)
Comment (przemyslawzalewski) [reply array/unique.json]: I have changed it to "How to find...".

File: async/README.md (hunk 10 above)
Comment (przemyslawzalewski) [reply async/README.md]: Thanks, this was an editing error. Fixed :)

File: debug/assert.test.ts (hunk 11 above)
Comment (przemyslawzalewski) [reply debug/assert.test.ts]: Too aggressive replacement. Thanks!

**Pull request merged**